packages/api/src/external/fhir/document/draft-update-document-reference.ts

```diff
@@ -1,74 +1,125 @@
+  const contained = getContained(inputDocRef, organization);
+  const extension = getExtensions(inputDocRef);
   const subject = toFHIRSubject(patientId);
-
-  const attachment: Attachment = {
-    title: "pre-upload placeholder",
-    url: s3Utils.buildFileUrl(bucketName, s3Key),
-  };
-
-  const content: DocumentReferenceContent[] = [{ attachment }];
-
+  const content = getContent(s3Key, s3BucketName);
   return {
     resourceType: "DocumentReference",
     id: docRefId,
-    contained: prelimDocRef.contained
-      ? [...prelimDocRef.contained, containedOrgWithOID]
-      : [containedOrgWithOID],
+    masterIdentifier,
+    identifier,
+    contained,
     status: "current",
     docStatus: "preliminary",
-    text: prelimDocRef.text ?? undefined,
-    extension: prelimDocRef.extension
-      ? [...prelimDocRef.extension, temporaryExtension]
-      : [temporaryExtension],
-    type: prelimDocRef.type,
-    category: prelimDocRef.category ?? undefined,
-    date: prelimDocRef.date ?? new Date().toISOString(),
-    author: prelimDocRef.author ?? undefined,
+    text: inputDocRef.text,
+    extension,
+    type: inputDocRef.type,
+    category: inputDocRef.category ?? undefined,
+    date: inputDocRef.date ?? new Date().toISOString(),
+    author: inputDocRef.author,
     subject,
-    description: prelimDocRef.description,
-    context: prelimDocRef.context,
+    description: inputDocRef.description,
+    context: inputDocRef.context,
     content,
   };
 }
+
+function getMasterIdentifier(docRefId: string): Identifier {
+  return {
+    system: IETF_URL,
+    value: docRefId,
+  };
+}
+
+function getIdentifiers(inputDocRef: DocumentReference): Identifier[] {
+  return [
+    ...(inputDocRef.masterIdentifier ? [inputDocRef.masterIdentifier] : []),
+    ...(inputDocRef.identifier ?? []),
+  ];
+}
+
+function getContained(inputDocRef: DocumentReference, organization: OrganizationModel): Resource[] {
+  const containedOrg = toFHIR(organization);
+  const containedOrgWithOID = appendIdentifierOID(organization, containedOrg);
+  return inputDocRef.contained
+    ? [...inputDocRef.contained, containedOrgWithOID]
+    : [containedOrgWithOID];
+}
+
+function getExtensions(inputDocRef: DocumentReference): Identifier[] {
+  if (!inputDocRef.extension) return [temporaryExtension];
+  const allowedExtensions = inputDocRef.extension.filter(isAllowedExtension);
+  return [...allowedExtensions, temporaryExtension];
+}
+
+function isAllowedExtension(e: Extension): boolean {
+  if (e.url && e.url.toUpperCase().includes(metriportDataSourceExtension.url)) return false;
+
+  if (!e.valueCoding) return true;
+
+  const system = e.valueCoding.system;
+  if (system && system.toUpperCase().includes(metriportDataSourceExtension.valueCoding.code)) {
+    return false;
+  }
+  return e.valueCoding.code !== metriportDataSourceExtension.valueCoding.code;
+}
+
+function getContent(s3Key: string, s3BucketName: string): DocumentReferenceContent[] {
```

Comment:
I think this should be named in more accordance with the "business operation"
so maybe "getAttachment"?
I understand its getting a DocumentReferenceContent object, so getContent makes sense from that perspective, but I got confused here and thought it was getting the actual Document Reference Contents as opposed to what is really just a URL

Reply:
There's an `attachment` element in FHIR, so that would be misleading.

I agree we should name by business logic/use case if we're talking about a high level function that implements/represents such a thing.

But this is a local function, very specific, so I think it should be named as specific as possible.